syntax = "proto3";
package auth.v1;

option go_package = "./auth;authv1";

service Authenticate {
  rpc Login (LoginReq) returns (LoginRes);
  rpc Refresh (RefreshReq) returns (RefreshRes);
  rpc Register(RegisterReq) returns (RegisterRes);
  rpc InActive(InActiveReq) returns (InActiveRes);
}

service AccountService {
    rpc CreateAccount (AccountCreateReq) returns (AccountRes);
    rpc GetAccount (AccountGetReq) returns (AccountRes);
    rpc UpdateAccount (AccountUpdateReq) returns (AccountRes);
    rpc DeleteAccount (AccountDelReq) returns (AccountDelRes);
}

message LoginReq {
  string login_id = 1;
  string password = 2;
}

message LoginRes {
  string access = 1;
  string refresh = 2;
}

message RefreshReq {
    string refresh = 1;
}

message RefreshRes {
    string access = 1;
}

message RegisterReq {
  string login_id = 1;
  string password = 2;
}

message RegisterRes {
  string access = 1;
  string refresh = 2;
}

message InActiveReq {
  string login_id = 1;
  string password = 2;
}

message InActiveRes {
  string message = 1;
}


message AccountCreateReq {
    string login_id = 1;
    string password = 2;
}

message AccountGetReq {
    string account_id = 1;
}

message AccountUpdateReq {
    string account_id = 1;
    string login_id = 2;
    string password = 3;
}

message AccountDelReq {
    string account_id = 1;
}

message AccountRes {
    string account_id = 1;
    string login_id = 2;
    string password = 3;
}

message AccountDelRes {
    bool success = 1;
}
